lib/pharos/ssh/remote_file.rb

```diff
@@ -53,7 +53,7 @@ def read
       # True if the file exists. Assumes a bash-like shell.
       # @return [Boolean]
       def exist?
-        @client.exec?("[ -e #{escaped_path} ]")
+        @client.exec!("(sudo true || exit 1) && (sudo test -e #{escaped_path} && echo true || exit 0)").strip == "true"
```

Comment:
No, the `sudo true && sudo test ...` is a bad idea...

Maybe use something like this to fail on `sudo` errors but trap the `test` error case:

```ruby
@client.exec!('sudo sh -c "test -e #{escaped_path} && echo true || echo false") == "true"` 
```

I'm fairly confident that `test`/`stat` itself is unlikely to fail without something else also breaking, so it should avoid false-positives.